airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/java/io/airbyte/cdk/integrations/destination/jdbc/JdbcBufferedConsumerFactory.java

```diff
@@ -103,11 +101,21 @@ private static Function<ConfiguredAirbyteStream, WriteConfig> toWriteConfig(
 
       final String defaultSchemaName = schemaRequired ? namingResolver.getIdentifier(config.get("schema").asText())
           : namingResolver.getIdentifier(config.get(JdbcUtils.DATABASE_KEY).asText());
+      // Method checks for v2
       final String outputSchema = getOutputSchema(abStream, defaultSchemaName, namingResolver);
-
       final String streamName = abStream.getName();
-      final String tableName = namingResolver.getRawTableName(streamName);
-      final String tmpTableName = namingResolver.getTmpTableName(streamName);
+      final String tableName;
+      final String tmpTableName;
+      // TODO: Should this be injected from outside ?
+      if (TypingAndDedupingFlag.isDestinationV2()) {
+        final var finalSchema = Optional.ofNullable(abStream.getNamespace()).orElse(defaultSchemaName);
+        final var rawName = StreamId.concatenateRawTableName(finalSchema, streamName);
```

Comment:
ideally we'd inject the parsedcatalog (or StreamId) into this method, instead of recalculating the raw table name here - would that be a lot of work?

i.e. the ParsedCatalog should be source of truth, in case we ever want to change naming conventions, etc.

Reply:
It might be a bit of work I want to pair it with config passed as typed object and only pass around what the method needs in terms of context, I had notes where I wrote that most places JsonNode config was sent around to get `config.get("schema").asText()`

Reply:
Created this https://github.com/airbytehq/airbyte/issues/33131